cpu-miner.c

```diff
@@ -468,12 +468,29 @@ static bool gbt_work_decode(const json_t *val, struct work *work)
 		le32enc((uint32_t *)(cbtx+37), 0xffffffff); /* prev txout index */
 		cbtx_size = 43;
 		/* BIP 34: height in coinbase */
-		for (n = work->height; n; n >>= 8) {
-			cbtx[cbtx_size++] = n & 0xff;
-			if (n < 0x100 && n >= 0x80)
-				cbtx[cbtx_size++] = 0;
+		if (work->height < 17) {
```

Comment:
The following should handle height 0 correctly. I doubt anybody will need it in practice, but getting it right won't hurt.
```suggestion
		if (work->height >= 1 && work->height <= 16) {
```